ansible_base/oauth2_provider/models/refresh_token.py

```diff
@@ -25,7 +29,58 @@ class Meta(oauth2_models.AbstractRefreshToken.Meta):
 
-    def save(self, *args, **kwargs):
+    def _has_non_timestamp_changes(self, update_fields=None):
+        """Check if any non-timestamp fields have changed.
+
+        Args:
+            update_fields: Optional list/set of field names being updated. If provided,
+                          only these fields will be checked for changes.
+
+        Returns:
+            bool: True if any non-timestamp field has changed, False otherwise.
+        """
         if not self.pk:
+            return False
+        try:
+            old_instance = OAuth2RefreshToken.objects.get(pk=self.pk)
+            # Fields to exclude from change detection (timestamp/auto-updated fields)
+            exclude_fields = {'created', 'created_by', 'modified', 'modified_by'}
+
+            # If update_fields is specified, only check those fields (minus excluded ones)
+            if update_fields is not None:
+                fields_to_check = set(update_fields) - exclude_fields
+                if not fields_to_check:
+                    # Only timestamp fields are being updated
+                    return False
+                # Check only the specified fields
+                for field_name in fields_to_check:
+                    old_value = getattr(old_instance, field_name, None)
+                    new_value = getattr(self, field_name, None)
+                    if old_value != new_value:
+                        return True
+            else:
+                # Check all non-timestamp fields
+                for field in self._meta.get_fields():
+                    if field.concrete and hasattr(field, 'name') and field.name not in exclude_fields:
+                        old_value = getattr(old_instance, field.name, None)
+                        new_value = getattr(self, field.name, None)
+                        if old_value != new_value:
+                            return True
+        except OAuth2RefreshToken.DoesNotExist:
+            pass
+        return False
+
+    def save(self, *args, **kwargs):
+        create_token = not bool(self.pk)
+        update_fields = kwargs.get('update_fields')
+        modifying_token = self._has_non_timestamp_changes(update_fields) if not create_token else False
+
+        if create_token:
             self.token = hash_string(self.token, hasher=hashlib.sha256, algo="sha256")
         super().save(*args, **kwargs)
+        access_token_id = self.access_token.pk if hasattr(self, 'access_token') and self.access_token else "N/A"
+        user_name = self.user.username if self.user else "N/A"
+        if create_token:
+            log_auth_event(f"Created OAuth2 refresh token {self.pk} for user '{user_name}' linked to access token {access_token_id}", second_logger=logger)
+        elif modifying_token:
+            log_auth_event(f"Modified OAuth2 refresh token {self.pk} for user '{user_name}' linked to access token {access_token_id}", second_logger=logger)
```

Comment:
If we make `_has_non_timestamp_changes` return true `if not self.pk` can simplify this to:

```suggestion
        update_fields = kwargs.get('update_fields')
        has_non_trivial_fields = self._has_non_timestamp_changes(update_fields)

        logging_verb = 'Modified'
        if not self.pk:
            self.token = hash_string(self.token, hasher=hashlib.sha256, algo="sha256")
            logging_verb = 'Created'
        super().save(*args, **kwargs)
        access_token_id = self.access_token.pk if hasattr(self, 'access_token') and self.access_token else "N/A"
        user_name = self.user.username if self.user else "N/A"
        if has_non_trivial_fields:
            log_auth_event(f"{logging_verb} OAuth2 refresh token {self.pk} for user '{user_name}' linked to access token {access_token_id}", second_logger=logger)
```